[AVAX] Banff 16 (1.9.16)

evlekht opened the pull request:
This PR adds all avaxs changes up to Banff 16 (1.9.16) version. It also has some small fixes and disables `AddPermissionlessValidatorTx`, cause it could add validators to primary network bypassing our validators logic.
## Co-authors
@mo-c4t github workflows, golangci
@havan .md files

StephenButtolph committed:
Support block.WithVerifyContext in chain.BlockWrapper (#2341)

Co-authored-by: Michael <michael@avalabs.org>

StephenButtolph committed:
Fix BatchedParseBlock over rpcchainvm (#2328)

xc-ava committed:
Add SECURITY.md (#2331)

aaronbuchwald committed:
Add Teleporter message format (#2180)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Stephen Buttolph <stephen@avalabs.org>

Commit:
Add Teleporter message signing to snow.Context (#2197)

Co-authored-by: Stephen Buttolph <stephen@avalabs.org>

StephenButtolph committed:
Improve subnetID lookup to support non-whitelisted subnets (#2354)

Commit:
Add Teleporter message verification (#2207)

Co-authored-by: Stephen <stephen@avalabs.org>

Commit:
Don't measure local message latency (#2356)

StephenButtolph committed:
Log poll responses before bubbling (#2357)

StephenButtolph committed:
Fix uintptr handling in codec.Marshal (#2360)

abi87 committed:
Add codec.Size (#2343)

Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Support non-whitelisted subnets in GetValidatorSet (#2369)

patrick-ogrady committed:
Add `UnpackLimitedBytes` and `UnpackLimitedStr` to `wrappers.Packer` (#2372)

aaronbuchwald committed:
Add await helpers for clients (#2359)

StephenButtolph committed:
Add `teleporter.AnycastID` (#2375)

StephenButtolph committed:
Add proposer list to proposer.Windower (#2366)

Co-authored-by: Patrick O'Grady <prohb125@gmail.com>

Commit:
Remove `testSettable` (#2364)

Commit:
Move bit sets to the set package (#2365)

StephenButtolph committed:
Support IP updates in PeerList gossip tracking (#2374)

Co-authored-by: Joshua Kim <20001595+joshua-kim@users.noreply.github.com>
Co-authored-by: Patrick O'Grady <prohb125@gmail.com>

joshua-kim committed:
Add initial network README (#2329)

StephenButtolph committed:
Add more wallet examples (#2382)

StephenButtolph committed:
Increase versions for v1.9.5 (#2383)

hexfusion committed:
Increase buf version to v1.11.0 (#2388)

StephenButtolph committed:
Add `fmt.Errorf` linter (#2391)

StephenButtolph committed:
Fix incorrect error handling (#2393)

StephenButtolph committed:
Fix staking disabled HealthChecks and connectivity (#2390)

Co-authored-by: aaronbuchwald <aaron.buchwald56@gmail.com>

StephenButtolph committed:
Add teleporter verification test with no validators (#2389)

abi87 committed:
Standardize `require` usage and remove `t.Fatal` from platformvm (#2297)

Co-authored-by: Joshua Kim <20001595+joshua-kim@users.noreply.github.com>
Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Minor cleanup with updated dependencies (#2397)

gyuho committed:
Pin goreleaser version to v1.13.1 (#2402)

coffeeavax committed:
Add block serialization format to the avm (#2095)

Co-authored-by: Stephen <stephen@avalabs.org>

abi87 committed:
Add dynamic state sync support (#2362)

Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Update auth API to comply with updated jwt specification (#2400)

StephenButtolph committed:
Use enums where possible in protos (#2404)

coffeeavax committed:
Add EngineType for ambiguous p2p messages (#2272)

Co-authored-by: Gyuho Lee <gyuho.lee@avalabs.org>
Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Remove dynamic constant error creation (#2392)

gyuho committed:
Add accepted frontier in Chits proto (#2106)

Co-authored-by: Stephen <stephen@avalabs.org>

gyuho committed:
Populate accepted frontier when sending chits (#2121)

Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Populate reasonable rpcchainvm logger in vm server (#2408)

gyuho committed:
Apply accepted frontier rather than failing a query (#2135)

Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Simplify minor P-chain state handling (#2412)

Commit:
add `List() []T` method to deque (#2403)

Commit:
Add Index method to deque (#2407)

joshua-kim committed:
Add average P2P connection duration metric (#2380)

Commit:
Fix typo `StateSummaryDynamic` to `StateSyncDynamic` (#2419)

StephenButtolph committed:
Add locked stakeable creation wallet example (#2417)

abi87 committed:
Remove proposervm height index reset (#2414)

StephenButtolph committed:
Remove usage of apt-key and unify install across versions (#2423)

joshua-kim committed:
Cleanup IP usage by embedding IPs (#2428)

StephenButtolph committed:
Fix NAT-PMP router discovery and port mapping (#2426)

patrick-ogrady committed:
Support ledger-avalanche@v0.6.5 (#2427)

Co-authored-by: Stephen Buttolph <stephen@avalabs.org>

StephenButtolph committed:
Improve message length parsing tests (#2430)

Commit:
Replace `--build-dir` with `--plugin-dir` (#1741)

Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Expose SetLevel on the Logger interface (#2443)

StephenButtolph committed:
Update DAGVM interface to support linearization (#2442)

StephenButtolph committed:
Add `--track-subnets` to replace `--whitelisted-subnets` (#2439)

joshua-kim committed:
Remove proposervm signer + cert from `snow.Context` (#2447)

StephenButtolph committed:
Update Keystore API to be disabled by default (#2449)

StephenButtolph committed:
Add simple API metrics (#2448)

StephenButtolph committed:
Fix consensus handler message logging (#2456)

StephenButtolph committed:
Update versions for v1.9.6 (#2440)

StephenButtolph committed:
Handle unrelated BLS key removal during subnet validator set lookback (#2460)

StephenButtolph committed:
Update versions for v1.9.7 (#2461)

StephenButtolph committed:
Remove plugin handling from AMI generation (#2464)

StephenButtolph committed:
Remove copy of unused plugins directory (#2466)

StephenButtolph committed:
Remove unused rpm packaging scripts (#2465)

StephenButtolph committed:
Add AVM `state.Chain` interface (#2411)

Co-authored-by: Chloe <99216251+coffeeavax@users.noreply.github.com>

StephenButtolph committed:
Simplify consensus context atomic values (#2473)

StephenButtolph committed:
Release lock before attempting to close a connection (#2481)

StephenButtolph committed:
Remove engine dependency from chain registrants (#2482)

StephenButtolph committed:
Add TCP proxy support (#2476)

StephenButtolph committed:
Removed unused field from chain handler log (#2484)

StephenButtolph committed:
Add engine type to consensus context (#2485)

StephenButtolph committed:
Initialize last accepted block to the genesis block, not the stop vertex (#2483)

abi87 committed:
Fix `validators.Contains` documentation typo (#2507)

ceyonur committed:
Report percentages for uptimes in `platform.getCurrentValidators` (#2471)

abi87 committed:
Optimize P-chain GetValidators when NodeIDs are provided (#2263)

StephenButtolph committed:
Fix `RemoveSubnetValidatorTx` Weight Diff Calculation (#2510)

dhrubabasu committed:
Use generic btree implementation (#2522)

Commit:
Convert caching to use generics (#1880)

Co-authored-by: Stephen Buttolph <stephen@avalabs.org>

abi87 committed:
Cache stakers in platformvm get validators APIs (#2499)

Co-authored-by: Stephen <stephen@avalabs.org>

abi87 committed:
Standardize `stakeAmount` and `weight` usage in P-chain APIs (#2524)

StephenButtolph committed:
Fix linting for the test manager (#2530)

dhrubabasu committed:
Remove `creationTxFee` from `info.GetTxFeeResponse` (#2529)

dhrubabasu committed:
Remove `address` from `platformvm.GetBalanceRequest` (#2531)

charlie-ava committed:
Test AMI builder during RCs (#2491)

joshua-kim committed:
Add p2p network client helper (#2511)

Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Remove `utils.CopyBytes` (#2546)

StephenButtolph committed:
Always provide App messages to the VM (#2545)

itinance committed:
Update rjeczalik/notify 0.9.2 -> 0.9.3

StephenButtolph committed:
Support `stakeAmount` in `addDelegator` and `addSubnetValidator` (#2552)

StephenButtolph committed:
Add LRU cache eviction callback (#2556)

Commit:
Document database invariant on putting empty keys (#2559)

dboehm-avalabs committed:
Add alpha path-based merkle-radix tree with state syncing (#1819)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Darioush Jalali <darioush.jalali@avalabs.org>
Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Update versions for v1.9.8 (#2557)

patrick-ogrady committed:
Remove noop `trieView.calculateIDs` trace spans (#2565)

StephenButtolph committed:
Cleanup database interface testing (#2566)

StephenButtolph committed:
Use set functions rather than map functions (#2580)

Commit:
Link to SECURITY.md from README.md (#1211)

Commit:
Limit number of concurrent hashing goroutines in merkledb (#2574)

Co-authored-by: Stephen Buttolph <stephen@avalabs.org>

StephenButtolph committed:
Add additional `Batch.Replay` invariant tests (#2567)

StephenButtolph committed:
Remove `nodb` and duplicate `Batch` code (#2569)

cam-schultz committed:
Add `IsProcessing` method to `chain.State` (#2578)

hexfusion committed:
Fix typo in `merkledb/README.md`  (#2589)

michaelkaplan13 committed:
Rename the `teleporter` package to `warp` (#2573)

Commit:
Fix flaky `TestDialerCancelDial` (#2590)

StephenButtolph committed:
Warn potentially publicly reachable http servers (#2592)

StephenButtolph committed:
Remove delegators from multiple nodeID getCurrentValidators calls (#2560)

Co-authored-by: Alberto Benegiamo <alberto.benegiamo@gmail.com>

Commit:
Add `StateSyncMinVersion` to `sync.ClientConfig` (#2572)

philipglazman committed:
Verify secp256k1 public key size during parse (#1215)

ceyonur committed:
Refactor subnets and subnet configs (#2472)

StephenButtolph committed:
Log a warning if the p2p IP is private during startup (#2599)

abi87 committed:
Validate genesis's `InitialStakeDuration` (#2542)

StephenButtolph committed:
Remove unused crypto tooling (#2607)

StephenButtolph committed:
Remove crypto package interfaces (#2609)

Co-authored-by: Dhruba Basu <7675102+dhrubabasu@users.noreply.github.com>

StephenButtolph committed:
Add rfc6979 compliance test vectors (#2611)

dhrubabasu committed:
Rename `TestEnvinronment` to `TestEnvironment` (#2618)

abi87 committed:
Unify P-chain UTXO handler interface (#2570)

abi87 committed:
Remove unused field `common.Config#Validators` (#2604)

Commit:
Remove reflect call when marshalling an empty slice (#2363)

Commit:
Fix `ip` config descriptions (#2614)

coffeeavax committed:
Add X-chain mempool data structure (#2562)

Co-authored-by: Stephen Buttolph <stephen@avalabs.org>

ceyonur committed:
Add `allowedNodes` for `validatorOnly` subnet configs (#2503)

abi87 committed:
Replace `bool` flags in P-chain state diffs with an enum (#2533)

hexfusion committed:
Remove go-plugin dependency (#2563)

Co-authored-by: Stephen <stephen@avalabs.org>

hexfusion committed:
Replace `ioCollector` with `io.Copy` (#2623)

Commit:
Replace id bags with generic implementations (#1879)

Co-authored-by: Stephen <stephen@avalabs.org>

dependabot[bot] committed:
Bump golang.org/x/net from 0.1.0 to 0.7.0 (#2622)

StephenButtolph committed:
Add additional comments to `block.WithVerifyContext` interface (#2568)

hexfusion committed:
Pass `GRPC_` and `GODEBUG` environments to subprocesses (#2628)

hexfusion committed:
Do not provide `os.Stdin` to a subprocess' `Stdin` (#2629)

Commit:
merkledb nit (#2632)

StephenButtolph committed:
Fix AMI IP resolution config (#2634)

StephenButtolph committed:
Update versions for `v1.9.9` (#2625)

StephenButtolph committed:
Remove `--whitelisted-subnets` (#2468)

dhrubabasu committed:
Update secp256k1 version to v4.1.0 (#1634)

Commit:
Simplify `app` package structure (#2631)

gyuho committed:
Rename Avalanche consensus metrics to `avalanche_{chainID}_avalanche` (#2588)

Commit:
Remove parents map from merkledb trieview (#2601)

Co-authored-by: David Boehm <91908103+dboehm-avalabs@users.noreply.github.com>

dboehm-avalabs committed:
Remove base caches from the trie view (#2640)

Commit:
Merkledb remove panic during cache eviction (#2638)

StephenButtolph committed:
Migrate utxo interfaces into `avax` package (#2643)

Co-authored-by: Chloe <99216251+coffeeavax@users.noreply.github.com>

StephenButtolph committed:
Pass VM configs rather than factories (#2645)

Co-authored-by: Chloe <99216251+coffeeavax@users.noreply.github.com>

StephenButtolph committed:
Remove `platformvm.VM#txExecutorBackend`  (#2648)

Co-authored-by: Alberto Benegiamo <alberto.benegiamo@gmail.com>

StephenButtolph committed:
Simplify VM Factory interface (#2649)

abi87 committed:
Move p-chain `Validator` and `SubnetValidator` to the `txs` package (#2636)

StephenButtolph committed:
Refactor X-chain API tx creation (#2654)

Co-authored-by: Chloe <99216251+coffeeavax@users.noreply.github.com>

Commit:
Remove `timer.TimeoutManager` (#2653)

Commit:
Use the `maps` package when possible (#2647)

hexfusion committed:
Simplfy subprocess `Stop` & remove useless test (#2652)

hexfusion committed:
Remove dangling buf.lock (#2659)

dboehm-avalabs committed:
Add value digests and value verification in proofs (#2639)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Darioush Jalali <darioush.jalali@avalabs.org>

patrick-ogrady committed:
Add `Signature.NumSigners` to Warp messages (#2665)

patrick-ogrady committed:
Standardize `set.Bits` method naming with `set.Bits64` (#2667)

StephenButtolph committed:
Use AVM config in `tx.SyntacticVerify` (#2666)

Co-authored-by: Chloe <99216251+coffeeavax@users.noreply.github.com>

StephenButtolph committed:
Add BLS public key to `snow.Context` (#2661)

StephenButtolph committed:
Update to go1.20.1 (#2671)

joshua-kim committed:
Set linger on p2p connections (#2646)

Co-authored-by: Patrick O'Grady <prohb125@gmail.com>

Commit:
[merkledb] remove `cache.Cache` from `onEvictCache` (#2674)

hexfusion committed:
Ignore sigint signals to ensure proper shutdown flow (#2682)

StephenButtolph committed:
Update versions for v1.9.10 (#2673)

Co-authored-by: Patrick O'Grady <prohb125@gmail.com>

StephenButtolph committed:
Remove unused linter from nolint (#2686)

Commit:
update readme (#2644)

dboehm-avalabs committed:
Validity tracking and Transactional Commit (#2642)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Darioush Jalali <darioush.jalali@avalabs.org>

StephenButtolph committed:
Drop writes to `NoLog` (#2698)

felipemadero committed:
Add ledger tx parsing support (#2695)

StephenButtolph committed:
Add vm-factory logger (#2701)

hexfusion committed:
Return correct initializer error (#2700)

StephenButtolph committed:
Update versions for v1.9.11 (#2702)

joshua-kim committed:
Revert "Set linger on p2p connections (#2646)" (#2703)

Commit:
Remove unused state package (#2694)

joshua-kim committed:
Switch noisy peer gossip log from debug to verbo (#2684)

joshua-kim committed:
Improve error message when failing to calculate subnet uptimes (#2683)

Commit:
Use linked hashmap in LRU cache (#2685)

charlie-ava committed:
Remove unused dir from deb packages (#2714)

StephenButtolph committed:
Export errors for X-chain testing (#2716)

StephenButtolph committed:
Unify spelling of serialized to the american format (#2717)

StephenButtolph committed:
Add X-chain create asset wallet example (#2715)

StephenButtolph committed:
Use `constants.UnitTestID` where possible (#2718)

StephenButtolph committed:
Add final state changes for the linearization (#2721)

StephenButtolph committed:
Remove block timer interface from mempool (#2720)

StephenButtolph committed:
Refactor X-chain tx syntactic verification (#2722)

StephenButtolph committed:
Remove unnecessary function call (#2725)

StephenButtolph committed:
Refactor X-chain tx semantic verification (#2723)

StephenButtolph committed:
Refactor X-chain tx execution (#2724)

Co-authored-by: Chloe <99216251+coffeeavax@users.noreply.github.com>

StephenButtolph committed:
Move `toEngine` channel into mempool interface (#2726)

StephenButtolph committed:
Move P-chain message format to components for use in X-chain (#2730)

Co-authored-by: Chloe <99216251+coffeeavax@users.noreply.github.com>

StephenButtolph committed:
Replace error strings with error objects in mempools (#2732)

StephenButtolph committed:
Deprecate various APIs (#2734)

Commit:
Add `warp.UnsignedMessage#ID` utility function (#2689)

aaronbuchwald committed:
Improve warp error messages (#2738)

StephenButtolph committed:
Fix order of errors in test (#2740)

Commit:
Use LinkedHashmap in the avm wallet service (#2737)

StephenButtolph committed:
Remove unused EngineTypes from p2p messages (#2735)

StephenButtolph committed:
Remove anonymous function calling handler.Consensus (#2736)

Co-authored-by: Joshua Kim <20001595+joshua-kim@users.noreply.github.com>

Commit:
Add non-blocking bounded queue (#2657)

joshua-kim committed:
Add support for dynamic consensus engine dispatching (#2581)

Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Add RemoveSubnetValidatorTx wallet example (#2743)

StephenButtolph committed:
Add X-chain block execution manager (#2727)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Chloe <99216251+coffeeavax@users.noreply.github.com>

StephenButtolph committed:
Drop all inbound messages after the stop vertex is accepted (#2746)

StephenButtolph committed:
Convert consensus+decision dispatchers into block+tx+vertex dispatchers (#2745)

StephenButtolph committed:
Mark X-chain migration as Cortina (#2747)

StephenButtolph committed:
Expose X-chain block parsing for dependencies (#2750)

dboehm-avalabs committed:
Locking rework (#2693)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Darioush Jalali <darioush.jalali@avalabs.org>

Commit:
Delete fundamentally racy test (#2752)

Co-authored-by: Stephen <stephen@avalabs.org>

Commit:
Separate publish docker image from kurtosis test workflow (#2710)

StephenButtolph committed:
Add unpopulated merkle root to X-chain blocks (#2751)

StephenButtolph committed:
Build avalanchego image before trying to publish (#2754)

StephenButtolph committed:
Add X-chain block builder (#2728)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Chloe <chloe.nguyen@avalabs.org>

StephenButtolph committed:
Add X-chain networking implementation (#2731)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Chloe <chloe.nguyen@avalabs.org>

StephenButtolph committed:
Add HTTP timeouts (#2705)

Co-authored-by: Joshua Kim <20001595+joshua-kim@users.noreply.github.com>

Commit:
Remove `context.TODO` from dynamic IP resolution (#2681)

StephenButtolph committed:
Add post linearization support to the X-chain APIs (#2708)

Commit:
Add `avm.GetBlock` API (#2755)

Co-authored-by: Stephen <stephen@avalabs.org>

Commit:
Add X-Chain `GetBlockByHeight` API method (#2757)

Commit:
Add X-Chain `GetHeight` API (#2758)

StephenButtolph committed:
Fix typo lineraized -> linearized (#2759)

StephenButtolph committed:
Start Snowman++ after Avalanche is Linearized (#2478)

StephenButtolph committed:
Reduce default maximum reconnect delay to 1 minute (#2762)

dboehm-avalabs committed:
Improve merkle proof locking (#2761)

gyuho committed:
Add X-chain txs metrics (#2756)

Co-authored-by: Stephen <stephen@avalabs.org>

Commit:
Provide Github Action Artifacts after builds (#2173)

StephenButtolph committed:
Fix Banff tx signing with ledger in the wallet (#2744)

StephenButtolph committed:
Push docker image on new tags (#2766)

StephenButtolph committed:
Transition to snowman bootstrapping immediately if StopVertex is accepted (#2765)

StephenButtolph committed:
Fix semver go version in CI (#2768)

abi87 committed:
Add validation status logging (#1236)

Co-authored-by: Stephen Buttolph <stephen@avalabs.org>

dboehm-avalabs committed:
[WIP] Attempt to fix sync test flakiness (#2760)

Co-authored-by: Darioush Jalali <darioush.jalali@avalabs.org>
Co-authored-by: Dan Laine <daniel.laine@avalabs.org>

dboehm-avalabs committed:
make value []byte that leave the library safe to edit (#2763)

Co-authored-by: Stephen Buttolph <stephen@avalabs.org>
Co-authored-by: Dan Laine <daniel.laine@avalabs.org>

StephenButtolph committed:
Update versions for v1.9.12 (#2767)

StephenButtolph committed:
Unblock X-chain bootstrapping (#2770)

aaronbuchwald committed:
Update versions for v1.9.13 (#2771)

StephenButtolph committed:
Drop X-chain indexing errors on accept (#2773)

aaronbuchwald committed:
Update versions for v1.9.14 (#2772)

dboehm-avalabs committed:
Fix error in Changeproof end key selection (#2776)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>

StephenButtolph committed:
Reduce optimal consensus processing limit to 10 (#2777)

StephenButtolph committed:
Add test for `verifyFxUsage` (#2690)

StephenButtolph committed:
Fix performance regression (#2774)

StephenButtolph committed:
Add example usage for the X-chain block indexer API (#2778)

Co-authored-by: Dan Laine <daniel.laine@avalabs.org>
Co-authored-by: Dhruba Basu <7675102+dhrubabasu@users.noreply.github.com>

Commit:
Update year in license header (#2742)

Co-authored-by: Stephen <stephen@avalabs.org>

StephenButtolph committed:
Update versions for v1.9.15 (#2779)

ceyonur committed:
Add errors to snowstorm graph lookups (#2791)

StephenButtolph committed:
Add additional snowstorm lookup check (#2790)

StephenButtolph committed:
Remove rejected vertices from frontier during same Avalanche traversal (#2796)

StephenButtolph committed:
Remove unnecessary repoll after vertex rejection (#2795)

Commit:
Reduce default maximum healthy processing time (#2784)

Commit:
Reduce default maximum healthy processing items (#2783)

StephenButtolph committed:
Canonicalize OperationTx (#2792)

StephenButtolph committed:
Update versions for v1.9.16 (#2793)

knikos reviewed (dismissed):
As far as conflicting changes and missed commits  are concerned, looks fine.

c4t-ag commented on vms/avm/asset_test.go:
```diff
@@ -11,7 +11,7 @@ import (
 )
 
-func TestGetAssetDescriptionC4T(t *testing.T) {
+func TestGetAssetDescriptionC4T(t *testing.T) { // TODO @evlekht do we still needed? That was removed by avax
```
Was it really removed? I wonder why it is not in service_test.go file (what would be the logic place for it IMO)
If getAssetDescription is not deprecated, I suggest that we move the test into our camino_service_test.

evlekht replied:
right, found avax test in service_test.go, our asset_test.go with avax license was added waaay back
I'll rename file to camino_service_test.go and fix license

c4t-ag commented on version/compatibility.json:
```diff
@@ -1,22 +1,6 @@
-  ],
-  "21": [
-    "v1.9.5"
+    "v0.4.10"
```
we should go to 0.5.0

c4t-ag replied:
Clearified: 0.4.10 is fine

c4t-ag commented on .github/workflows/publish_docker_image.yml:
```diff
@@ -20,3 +20,4 @@ jobs:
           DOCKER_PASS: ${{ secrets.docker_pass }}
           DOCKER_REPO: ${{ secrets.docker_repo }}
         run: .github/workflows/publish_image.sh
+// TODO@ pure avax copy, do we need it?
```
IMO no -> remove

evlekht replied:
thats outdated, was addressed in some of fixes commits

c4t-ag commented on .github/workflows/static-analysis.yml:
```diff
@@ -20,3 +20,4 @@ jobs:
       - name: Run static analysis tests
         shell: bash
         run: scripts/lint.sh
+// TODO@ avax renamed, rename back?
```
I don't care -> your decision :-)

evlekht replied:
thats outdated, was addressed in some of fixes commits

c4t-ag commented on SECURITY.md:
```diff
@@ -15,3 +15,4 @@ Please refer to the [Bug Bounty Page](https://hackenproof.com/avalanche) for the
 ## Supported Versions
 
 Please use the [most recently released version](https://github.com/ava-labs/avalanchego/releases/latest) to perform testing and to validate security issues.
+// TODO@ pure avax copy, update with camino or remove
```
Will be more important if we finally merge camino-node and caminogo -> I suggest keep / rename

evlekht replied:
thats outdated, was addressed in some of fixes commits

c4t-ag commented on .golangci.yml:
```diff
@@ -85,8 +85,8 @@ linters-settings:
-          - "rand.Read"
-          - "sb.WriteString"
+          - "math/rand\\.Read"
+          - "strings\\.Builder\\.WriteString"
```
Whats the reason for this / and \\ stuff here ?

evlekht replied:
@mo-c4t

mo-c4t replied:
this was added by avax. to add some functions to that rule because they are returning errors while linting.  I believe "\\." is the same as "." but / has other meaning, I think the rand function under math lib. and the / is  needed here.

c4t-ag commented on vms/platformvm/camino_vm_test.go:
```diff
@@ -476,7 +476,7 @@ func TestDepositsAutoUnlock(t *testing.T) {
 	}})
 	vm.ctx.Lock.Lock()
-	defer func() { require.NoError(vm.Shutdown(context.Background())) }() //nolint:revive
+	defer func() { require.NoError(vm.Shutdown(context.Background())) }() //nolint:lint
```
Sure? Reading the error I expected that the rule can be dropped completely (??)

evlekht replied:
@mo-c4t

mo-c4t replied:
after updating the golangci-lint version, the error comes with nolint:lint and not nolint:revive

evlekht committed:
[AVAX] Avalanche Banff 16 (1.9.16)

Merge branch '1.9.16-avax' into 1.9.16-avax-to-dev

# Conflicts:
#	.github/ISSUE_TEMPLATE/bug_report.md
#	.github/packer/roles/public-ami/defaults/main.yml
#	.github/packer/roles/public-ami/tasks/main.yml
#	.github/packer/roles/public-ami/templates/conf.json.j2
#	.github/packer/ubuntu-focal-x86_64-public-ami.json
#	.github/workflows/buf-lint.yml
#	.github/workflows/build-and-test-mac-windows.yml
#	.github/workflows/build-and-test.yml
#	.github/workflows/build-deb-pkg.sh
#	.github/workflows/build-linux-binaries.yml
#	.github/workflows/build-macos-release.yml
#	.github/workflows/build-public-api.yml
#	.github/workflows/build-tgz-pkg.sh
#	.github/workflows/build-ubuntu-amd64-release.yml
#	.github/workflows/build-ubuntu-arm64-release.yml
#	.github/workflows/build-win-release.yml
#	.github/workflows/build.yml
#	.github/workflows/publish_image.sh
#	.github/workflows/run-kurtosis-tests-basic.yml
#	.github/workflows/run-kurtosis-tests-full.yml
#	.github/workflows/run_e2e_tests.sh
#	.github/workflows/static-analysis.yml
#	.github/workflows/test.e2e.yml
#	.github/workflows/test.upgrade.yml
#	.github/workflows/update-ami.py
#	Dockerfile
#	README.md
#	RELEASES.md
#	api/admin/client.go
#	api/admin/service.go
#	api/info/client.go
#	api/info/service.go
#	app/app.go
#	config/config.go
#	config/flags.go
#	database/linkeddb/linkeddb.go
#	genesis/beacons.go
#	genesis/config.go
#	genesis/genesis.go
#	genesis/genesis_fuji.go
#	genesis/genesis_local.go
#	genesis/genesis_mainnet.go
#	genesis/genesis_test.go
#	genesis/params.go
#	genesis/unparsed_config.go
#	go.mod
#	go.sum
#	main/main.go
#	network/certs_test.go
#	network/network.go
#	network/peer/mock_gossip_tracker.go
#	network/peer/peer_test.go
#	network/peer/upgrader.go
#	node/beacon_manager.go
#	node/beacon_manager_test.go
#	node/insecure_validator_manager.go
#	node/node.go
#	scripts/build.sh
#	scripts/build_avalanche.sh
#	scripts/build_coreth.sh
#	scripts/build_image.sh
#	scripts/build_local_dep_image.sh
#	scripts/build_local_image.sh
#	scripts/build_test.sh
#	scripts/constants.sh
#	scripts/lint.sh
#	scripts/local.Dockerfile
#	scripts/mocks.mockgen.txt
#	scripts/tests.e2e.sh
#	scripts/tests.upgrade.sh
#	snow/networking/sender/mock_external_sender.go
#	staking/tls.go
#	tests/colors.go
#	tests/e2e/banff/suites.go
#	tests/e2e/describe.go
#	tests/e2e/e2e.go
#	tests/e2e/e2e_test.go
#	tests/e2e/p/permissionless_subnets.go
#	tests/e2e/p/workflow.go
#	tests/e2e/ping/suites.go
#	tests/e2e/static-handlers/suites.go
#	tests/e2e/x/transfer/virtuous.go
#	tests/e2e/x/whitelist-vtx/suites.go
#	tests/upgrade/upgrade_test.go
#	utils/constants/application.go
#	utils/constants/network_ids.go
#	utils/constants/network_ids_test.go
#	utils/crypto/bls/bls_test.go
#	utils/crypto/bls/public.go
#	utils/crypto/bls/public_test.go
#	utils/crypto/bls/secret.go
#	utils/crypto/bls/secret_test.go
#	utils/crypto/bls/signature.go
#	utils/crypto/bls/signature_test.go
#	utils/crypto/keychain/keychain.go
#	utils/crypto/keychain/keychain_test.go
#	version/constants.go
#	version/string.go
#	vms/avm/service.go
#	vms/components/avax/atomic_utxos.go
#	vms/platformvm/api/static_service.go
#	vms/platformvm/blocks/builder/builder.go
#	vms/platformvm/blocks/builder/builder_test.go
#	vms/platformvm/blocks/codec.go
#	vms/platformvm/blocks/executor/proposal_block_test.go
#	vms/platformvm/blocks/executor/standard_block_test.go
#	vms/platformvm/blocks/executor/verifier.go
#	vms/platformvm/blocks/executor/verifier_test.go
#	vms/platformvm/client.go
#	vms/platformvm/config/config.go
#	vms/platformvm/fx/fx.go
#	vms/platformvm/genesis/genesis.go
#	vms/platformvm/metrics/block_metrics.go
#	vms/platformvm/service.go
#	vms/platformvm/service_test.go
#	vms/platformvm/signer/empty.go
#	vms/platformvm/signer/empty_test.go
#	vms/platformvm/signer/proof_of_possession.go
#	vms/platformvm/signer/proof_of_possession_test.go
#	vms/platformvm/signer/signer.go
#	vms/platformvm/state/diff.go
#	vms/platformvm/state/mock_state.go
#	vms/platformvm/state/state.go
#	vms/platformvm/txs/codec.go
#	vms/platformvm/txs/executor/atomic_tx_executor.go
#	vms/platformvm/txs/executor/proposal_tx_executor.go
#	vms/platformvm/txs/executor/state_changes.go
#	vms/platformvm/txs/executor/tx_mempool_verifier.go
#	vms/platformvm/txs/visitor.go
#	vms/platformvm/utxo/handler.go
#	vms/platformvm/utxo/mock_verifier.go
#	vms/platformvm/vm.go
#	vms/proposervm/batched_vm_test.go
#	vms/proposervm/block.go
#	vms/proposervm/block/block.go
#	vms/proposervm/block/build.go
#	vms/proposervm/block/build_test.go
#	vms/proposervm/block/parse_test.go
#	vms/proposervm/post_fork_block_test.go
#	vms/proposervm/pre_fork_block_test.go
#	vms/proposervm/state/block_state_test.go
#	vms/proposervm/state_syncable_vm_test.go
#	vms/proposervm/vm_test.go
#	wallet/chain/p/context.go
#	wallet/chain/x/context.go
